### PR TITLE
[IMP] web: xml parsing doesn't fail silently

### PR DESCRIPTION
**Pull request by SimonGenin**

### CODE REVIEW

File: addons/web/static/tests/core/utils/xml_tests.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/** @odoo-module **/
+import { XMLParser } from "@web/core/utils/xml";
+
+QUnit.module("utils", () => {
+    QUnit.module("xml");
+
+    QUnit.test("parse error throws an exeption", async (assert) => {
```
Comment (aab-odoo) [inline addons/web/static/tests/core/utils/xml_tests.js:7]: exception

File: addons/web/static/src/core/utils/xml.js [hunk 2]
```diff
@@ -30,6 +30,11 @@ export class XMLParser {
     parseXML(arch) {
         const parser = new DOMParser();
         const xml = parser.parseFromString(arch, "text/xml");
+
+        if (xml.body && xml.body.firstChild && xml.body.firstChild.tagName === "parsererror") {
+            throw new Error(`An error occured while parsing ${arch}: ${xml.body.firstChild}`);
```
Comment (aab-odoo) [inline addons/web/static/src/core/utils/xml.js:35]: maybe check https://newbedev.com/how-do-i-detect-xml-parsing-errors-when-using-javascript-s-domparser-in-a-cross-browser-way

File: addons/web/static/tests/views/graph_view_tests.js [hunk 3]
```diff
@@ -1735,7 +1735,7 @@ QUnit.module("Views", (hooks) => {
             arch: `
                 <graph type="pie">
                     <field name="product_id"/>
-                <graph/>
+                </graph>
```
Comment (SimonGenin) [inline addons/web/static/tests/views/graph_view_tests.js:1738]: Found thanks to this PR :D

File: addons/web/static/src/core/utils/xml.js [hunk 4]
```diff
@@ -1,5 +1,21 @@
 /** @odoo-module **/
 
+// Both variables are used by the hasParsingError function.
+const errorneousParse = new DOMParser().parseFromString("<", "text/xml");
+const parsererrorNS = errorneousParse.getElementsByTagName("parsererror")[0].namespaceURI;
```
Comment (aab-odoo) [inline addons/web/static/src/core/utils/xml.js:5]: erroneous

Comment (aab-odoo) [reply addons/web/static/src/core/utils/xml.js:5]: this line triggers an error on firefox.

Note that "a valid xml with a <parsererror> tag" isn't a usecase we need to handle, so I'd go for `invalidXml.getElementsByTagName("parsererror")[0]`

File: addons/web/static/src/core/utils/xml.js [hunk 5]
```diff
@@ -1,5 +1,9 @@
 /** @odoo-module **/
 
+export function hasParsingError(parsedDocument) {
```
Comment (aab-odoo) [inline addons/web/static/src/core/utils/xml.js:3]: no need to export it

**Review by aab-odoo — APPROVED** [review]

File: addons/web/static/src/core/utils/xml.js (hunk 5 above)
Comment (SimonGenin) [reply addons/web/static/src/core/utils/xml.js]: @aab-odoo fixed

**Comment by aab-odoo** [comment]
robodoo r+

**Pull request closed without merging**